src/init.cpp

```diff
@@ -443,11 +443,19 @@ void SetupServerArgs()
     gArgs.AddArg("-zmqpubhashtx=<address>", "Enable publish hash transaction in <address>", false, OptionsCategory::ZMQ);
     gArgs.AddArg("-zmqpubrawblock=<address>", "Enable publish raw block in <address>", false, OptionsCategory::ZMQ);
     gArgs.AddArg("-zmqpubrawtx=<address>", "Enable publish raw transaction in <address>", false, OptionsCategory::ZMQ);
+    gArgs.AddArg("-zmqpubhashblockhwm=<n>", strprintf("Set publish hash block outbound message high water mark (default: %d)", DEFAULT_ZMQ_SNDHWM), false, OptionsCategory::ZMQ);
+    gArgs.AddArg("-zmqpubhashtxhwm=<n>", strprintf("Set publish hash transaction outbound message high water mark (default: %d)", DEFAULT_ZMQ_SNDHWM), false, OptionsCategory::ZMQ);
+    gArgs.AddArg("-zmqpubrawblockhwm=<n>", strprintf("Set publish raw block outbound message high water mark (default: %d)", DEFAULT_ZMQ_SNDHWM), false, OptionsCategory::ZMQ);
+    gArgs.AddArg("-zmqpubrawtxhwm=<n>", strprintf("Set publish raw transaction outbound message high water mark (default: %d)", DEFAULT_ZMQ_SNDHWM), false, OptionsCategory::ZMQ);
```

Comment:
I don't really like the duplication here (one argument for each publisher cross (address, hwm)), but I think it is probably the easiest solution.  At least I don't see a good alternative that would not be over-engineered if we want the HWM configurable for each publisher separately.

Reply:
yeah, i figured that at least it fit a pattern and was easy to code and review. unless someone has a better idea, i think i'll leave it simple.

Reply:
Maybe one flag to set them all would be interesting?

Reply:
that was how it was originally done in the other PR and then someone wanted it broken out i believe. it makes sense to me to have them separate because the tx publishes definitely need way more message capacity than the block pubs.

Reply:
Perhaps there could be just two (not four) HWMs, one for tx (raw and hash) and one for blocks?  (But that's just a thought - having all four is totally fine for me if someone has use for separate values.)

Reply:
there is still a big potential message size difference between just hash and raw{block,tx} so i think having them separate still is better. it just gives people the most flexibility. btw, thanks for the review!

Reply:
`-zmqpubrawtx=<address>;hwm=<hwm>` perhaps?

Reply:
@luke-jr FYI https://github.com/bitcoin/bitcoin/pull/14060#issuecomment-416385997